pkg/sdk/external_functions_impl_gen.go

```diff
@@ -0,0 +1,166 @@
+package sdk
+
+import (
+	"context"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+)
+
+var _ ExternalFunctions = (*externalFunctions)(nil)
+
+type externalFunctions struct {
+	client *Client
+}
+
+func (v *externalFunctions) Create(ctx context.Context, request *CreateExternalFunctionRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (v *externalFunctions) Alter(ctx context.Context, request *AlterExternalFunctionRequest) error {
+	opts := request.toOpts()
+	return validateAndExec(v.client, ctx, opts)
+}
+
+func (v *externalFunctions) Show(ctx context.Context, request *ShowExternalFunctionRequest) ([]ExternalFunction, error) {
+	opts := request.toOpts()
+	dbRows, err := validateAndQuery[externalFunctionRow](v.client, ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	resultList := convertRows[externalFunctionRow, ExternalFunction](dbRows)
+	return resultList, nil
+}
+
+func (v *externalFunctions) ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*ExternalFunction, error) {
+	externalFunctions, err := v.Show(ctx, NewShowExternalFunctionRequest().WithLike(&Like{Pattern: String(id.Name())}))
+	if err != nil {
+		return nil, err
+	}
+	return collections.FindOne(externalFunctions, func(r ExternalFunction) bool { return r.Name == id.Name() })
```

Comment:
To narrow down the search maybe add comparisons of `r.DatabaseName == id.DatabaseName() && r.SchemaName == id.SchemaName()`. We should also add here a comparison of the arguments because you can define two external functions with the same names, but different sets of arguments. Name + Args have to be unique in a given schema.

Reply:
fixed, yeah thats a good point, i realized this when I was working on adding procedures to sdk